lightspeed_stack_providers/providers/inline/safety/lightspeed_redaction/lightspeed_redaction/redaction.py

```
@@ -0,0 +1,211 @@
+        redacted_content: str = content
+        applied_rules: list[str] = []
+
+        for rule in self.compiled_rules:
+            try:
+                if rule["pattern"].search(redacted_content):
+                    redacted_content = rule["pattern"].sub(
+                        rule["replacement"], redacted_content
+                    )
+                    applied_rules.append(rule["original_pattern"])
+
+            except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)